may-2018-reenabled-opcodes.md

```diff
@@ -0,0 +1,428 @@
+* number of elements on stack is reduced by one
+
+Unit tests:
+
+1. `x1 x2 OP_AND -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_AND -> x1 & x2`. Check valid results.
+
+### OP_OR
+
+    Opcode (decimal): 133
+    Opcode (hex): 0x85
+
+Boolean *or* between each bit in the operands.
+
+	x1 x2 OP_OR -> out
+	
+The operator must fail if:
+1. `len(x1) != len(x2)`. The two operands must be the same size.
+
+Impact of successful execution:
+* stack memory use reduced by `len(x1)`
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `x1 x2 OP_OR -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_OR -> x1 | x2`. Check valid results.
+
+### OP_XOR
+
+    Opcode (decimal): 134
+    Opcode (hex): 0x86
+
+Boolean *xor* between each bit in the operands.
+
+	x1 x2 OP_XOR -> out
+	
+The operator must fail if:
+1. `len(x1) != len(x2)`. The two operands must be the same size.
+
+Impact of successful execution:
+* stack memory use reduced by `len(x1)`
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `x1 x2 OP_XOR -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_XOR -> x1 xor x2`. Check valid results.
+    
+## Arithmetic
+
+#### Note about canonical form and floor division
+
+Operands for all arithmetic operations are assumed to be numeric values and must be in canonical form.  
+See [data types](#data-types) for more information.
+
+**Floor division**
+
+Note: that when considering integer division and modulo operations with negative operands the rules applied in the C language and most
+languages (with Python being a notable exception) differ from the strict mathematical definition.  Script follows the C language set of
+rules.  Namely:
+1. Non-integer quotients are rounded towards zero
+2. The equation `(a/b)*b + a%b == a` is satisfied by the results
+3. From the above equation it follows that: `a%b == a - (a/b)*b`
+4. In practice if `a` is negative for the modulo operator the result will be negative or zero.
+
+
+### OP_DIV
+
+    Opcode (decimal): 150
+    Opcode (hex): 0x96
+    
+Return the integer quotient of `a` and `b`.  If the result would be a non-integer it is rounded *towards* zero.
+
+    a b OP_DIV -> out
+    
+    where a and b are interpreted as numeric values
+    
+The operator must fail if:
+1. `!isnum(a) || !isnum(b)`. Fail if either operand is not a numeric value.
+1. `b == 0`. Fail if `b` is equal to any type of zero.
+
+Impact of successful execution:
+* stack memory use reduced
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `a b OP_DIV -> failure` where `!isnum(a)` or `!isnum(b)`. Both operands must be numeric values
+2. `a 0 OP_DIV -> failure`. Division by positive zero (all sizes), negative zero (all sizes), `OP_0` 
+4. `27 7 OP_DIV -> 3`, `27 -7 OP_DIV -> -3`, `-27 7 OP_DIV -> -3`, `-27 -7 OP_DIV -> 3`. Check negative operands. 
```

Comment:
Test Case 3 is missing. It should probably be a case for correct integer division without remainders. For example:

`27 3 OP_DIV -> 9`, `27 -3 OP_DIV -> -9`, `-27 3 OP_DIV -> -9`, `-27 -3 OP_DIV -> 9`

Reply:
markdown renumbers as required, but updated numbers. successful tests covered by case 5 (really 4) below